tests/drivers/watchdog/wdt_error_cases/src/main.c

```diff
@@ -0,0 +1,1001 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/ztest_error_hook.h>
+
+/*
+ * To use this test, either the devicetree's /aliases must have a
+ * 'watchdog0' property, or one of the following watchdog compatibles
+ * must have an enabled node.
+ */
+#if DT_NODE_HAS_STATUS(DT_ALIAS(watchdog0), okay)
+#define WDT_NODE DT_ALIAS(watchdog0)
+#elif DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_wdt)
+#define WDT_NODE DT_INST(0, nordic_nrf_wdt)
+#elif DT_HAS_COMPAT_STATUS_OKAY(zephyr_counter_watchdog)
+#define WDT_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(zephyr_counter_watchdog)
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_CHOSEN(zephyr_dtcm), okay)
+#define NOINIT_SECTION ".dtcm_noinit.test_wdt"
+#else
+#define NOINIT_SECTION ".noinit.test_wdt"
+#endif
+
+/* Common for all targets: */
+#define DEFAULT_WINDOW_MAX	 (500U)
+#define DEFAULT_WINDOW_MIN	 (0U)
+#define DEFAULT_OPTIONS		 (WDT_OPT_PAUSE_IN_SLEEP | WDT_OPT_PAUSE_HALTED_BY_DBG)
+#define MAX_INSTALLABLE_TIMEOUTS (8)
+
+/* By default run all error checks,
+ * unset below to align test scope for the specific target
+ */
+#define WDT_FLAG_RESET_NONE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_CPU_CORE_IS_NOT_SUPPORTED
+#define WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#define WATCHDOG_SUPPORTS_ONLY_ONE_TIMEOUT_VALUE
+#define WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#define WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#define WDT_DISABLE_IS_NOT_SUPPORTED
+#define WDT_FEED_CAN_STALL
+
+/* Align tests to the specific target: */
+#if defined(CONFIG_SOC_NRF54L15)
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0x07CFFFFFU)
+#undef WDT_FLAG_RESET_SOC_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_IN_SLEEP_IS_NOT_SUPPORTED
+#undef WDT_OPT_PAUSE_HALTED_BY_DBG_IS_NOT_SUPPORTED
+#undef WDT_DISABLE_IS_NOT_SUPPORTED
+#undef WDT_FEED_CAN_STALL
+#else
+#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
+#define WDT_WINDOW_MAX_ALLOWED (0xFFFFFFFFU)
+#endif
```

Comment:
Using negated language here in combination with the `#undef`s and `#if !defined()` and `#if defined()` further down below makes it quite difficult to follow which features are actually tested for which MCU.

Could we instead define the flags with positive terms and use one `#define` which contains all the test flags?

As an example:

```suggestion
/* By default run all error checks,
 * unset below to align test scope for the specific target
 */
#define WDT_FLAG_RESET_NONE_SUPPORTED BIT(0)
#define WDT_FLAG_RESET_CPU_CORE_SUPPORTED BIT(1)
#define WDT_FLAG_RESET_SOC_SUPPORTED BIT(2)
#define WDT_FLAG_ONLY_ONE_TIMEOUT_VALUE_SUPPORTED BIT(3)
#define WDT_OPT_PAUSE_IN_SLEEP_SUPPORTED BIT(4)
#define WDT_OPT_PAUSE_HALTED_BY_DBG_SUPPORTED BIT(5)
#define WDT_DISABLE_SUPPORTED BIT(6)
#define WDT_FEED_CAN_STALL BIT(7)

/* Align tests to the specific target: */
#if defined(CONFIG_SOC_NRF54L15)
#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
#define WDT_WINDOW_MAX_ALLOWED (0x07CFFFFFU)
#define WDT_TEST_FLAGS (WDT_FLAG_RESET_CPU_CORE_SUPPORTED | ...)
#else
#define DEFAULT_FLAGS	       (WDT_FLAG_RESET_SOC)
#define WDT_WINDOW_MAX_ALLOWED (0xFFFFFFFFU)
#endif
```

Reply:
I made changes as You suggested. Let me know if this is what You intended.